dace/codegen/targets/fpga.py

```diff
@@ -677,6 +685,41 @@ def _emit_copy(self, sdfg, state_id, src_node, src_storage, dst_node,
 
             ctype = src_node.desc(sdfg).dtype.ctype
 
+            # For a single vector access, tolerate not having set a range
+            if copy_shape[-1] != 1:
+                # Adjust for vectorization length
+                copy_shape[-1] = copy_shape[-1] / memlet.veclen
```

Comment:
It's kinda weird that (a) we have a special case for size 1 and (b) we need to divide by memlet veclen but nothing to do I guess...

Also, maybe integer division? //